### PR TITLE
Add servers to slugs for Android

### LINKED ISSUE #163: Android automated tests do not support experiments in Preview
### Workflow commands Android:
- Branch: ['control']
- Experiment Slug: font-fingerprinting-protection-on-android
- Firefox Version: ['130.0']
- Additional Arguments; N/A

Testing with a Production experiment in Preview, a large portion of the tests failed, and all of them contained the nimbus-cli command missing the "preview/" prefix before the slug.

Android Test Run with report: https://github.com/mozilla/klaatu/actions/runs/10883728153

### PR DESCRIPTION
**Pull request by b4handjr**
This add the capability to choose an experimenter server to get an experiment to test against.

Fixes #163

### CODE REVIEW

File: .github/workflows/ios_manual.yml [hunk 1]
```diff
@@ -92,7 +101,8 @@ jobs:
           xcrun simctl shutdown all
           export SIMULATOR_UDID=$(python get_specific_device_udid.py)
           poetry run python generate_smoke_tests.py
-          poetry run pytest --experiment ${{ inputs.slug }} --experiment-branch ${{ matrix.branch }} --feature ${{ inputs.feature-name }} ${{ inputs.extra-arguments }}
+          echo "TEST COMMAND: poetry run pytest --experiment ${{inputs.experiment-server}}/${{ inputs.slug }} --experiment-branch ${{ matrix.branch }} --feature ${{ inputs.feature-name }} ${{ inputs.extra-arguments }}"
```
Comment (jaredlockhart) [inline .github/workflows/ios_manual.yml:104]: How come the formatting is different here than the Android one? Ie server/slug

File: tests/android/conftest.py [hunk 2]
```diff
@@ -132,10 +132,11 @@ def fixture_experiment_url(request, variables):
...
         # Build URL from slug
-        if request.config.getoption("--stage"):
-            url = f"{variables['urls']['stage_server']}/api/v6/experiments/{slug}/"
-        else:
-            url = f"{variables['urls']['prod_server']}/api/v6/experiments/{slug}/"
+        match request.config.getoption("--experiment-server"):
+            case "prod":
+                url = f"{variables['urls']['prod_server']}/api/v6/experiments/{slug}/"
+            case "stage" | "stage/preview":
+                url = f"{variables['urls']['stage_server']}/api/v6/experiments/{slug}/"
```
Comment (jaredlockhart) [inline tests/android/conftest.py:139]: Match!! 🫨

**Review by jaredlockhart — APPROVED** [review]
Looks like a lot of complexity is simplified.  Very nice 👍

File: .github/workflows/ios_manual.yml (hunk 1 above)
Comment (b4handjr) [reply .github/workflows/ios_manual.yml:104]: Oh good catch, this shouldn't be in this PR

**Pull request merged**